### ISSUE TITLE
"Service %s/spawn_entity unavailable." error.

### ISSUE THREAD

**Issue by serdonke** [eYantra-Robotics-Competition/eYRC-2023_Cosmo_Logistic#1]
The robot does not launch even after properly following all the instructions in the [TaskBook](https://portal.e-yantra.org/themeBook/cl/Task_0/warehouse_setup_task/task_instruction.html).

I sourced the `setup.bash` in gazebo-11 and sourced the setup config in the install directory after building.

Using this command to launch `ros2 launch ebot_description ebot_gazebo_launch.py`.

I get the following error.

```
[spawn_entity.py-4]   warnings.warn(
[spawn_entity.py-4] [INFO] [1693940883.576706852] [ebot_spawner]: Waiting for entity xml on robot_description_ebot
[spawn_entity.py-4] [INFO] [1693940883.587518073] [ebot_spawner]: Waiting for service /spawn_entity, timeout = 30
[spawn_entity.py-4] [INFO] [1693940883.587716692] [ebot_spawner]: Waiting for service /spawn_entity
[spawn_entity.py-4] [ERROR] [1693940913.654108778] [ebot_spawner]: Service %s/spawn_entity unavailable. Was Gazebo started with GazeboRosFactory?
[spawn_entity.py-4] [ERROR] [1693940913.654729433] [ebot_spawner]: Spawn service failed. Exiting.
[ERROR] [spawn_entity.py-4]: process has died [pid 9217, exit code 1, cmd '/opt/ros/humble/lib/gazebo_ros/spawn_entity.py -entity ebot -topic robot_description_ebot --ros-args -r __node:=ebot_spawner'].
```
Searching for the issue I stumbled on this https://answers.ros.org/question/374296/was-gazebo-started-with-gazeborosfactory-error-when-launching-urdf-file/

But it seems like I do not need to worry about that since

```
^C[WARNING] [launch]: user interrupted with ctrl-c (SIGINT)
[robot_state_publisher-3] [INFO] [1693940920.624595508] [rclcpp]: signal_handler(signum=2)
[gzserver-1] [INFO] [1693940920.624876671] [rclcpp]: signal_handler(signum=2)
[ERROR] [gzclient-2]: process has died [pid 9213, exit code -2, cmd 'gzclient'].
[INFO] [robot_state_publisher-3]: process has finished cleanly [pid 9215]
[ERROR] [gzserver-1]: process[gzserver-1] failed to terminate '5' seconds after receiving 'SIGINT', escalating to 'SIGTERM'
[INFO] [gzserver-1]: sending signal 'SIGTERM' to process[gzserver-1]
[gzserver-1] [INFO] [1693940925.631072407] [rclcpp]: signal_handler(signum=15)
[ERROR] [gzserver-1]: process[gzserver-1] failed to terminate '10.0' seconds after receiving 'SIGTERM', escalating to 'SIGKILL'
[INFO] [gzserver-1]: sending signal 'SIGKILL' to process[gzserver-1]
[ERROR] [gzserver-1]: process has died [pid 9211, exit code -9, cmd 'gzserver /home/patricio/pat/repos/task0_ws/install/eyantra_warehouse/share/eyantra_warehouse/worlds/eyantra_warehouse.world -slibgazebo_ros_init.so -slibgazebo_ros_factory.so -slibgazebo_ros_force_system.so'].
```

I installed everything mentioned in the `requirements.sh` manually this time and it still does not work.

**Comment by serdonke** [comment]
Things I have tried:
- Manually installing everything from within `requirements.sh`
- Installing all the suggested packages when installing the `requirements.sh` stuff.
- REbuilding, relaunching and doing everything over.

**Comment by serdonke** [comment]
Ok, some progress.

Gazebo now launches...if it is already running that is.

There is some problem with the staff's instructions I think because I have tried every possible thing and it still is not working.

**Comment by architjain19** [comment]
This is an issue due to your `gzserver` not responding correctly. For troubleshooting this, try killing your gzserver first and relaunch everything in a new terminal as some of your terminals might be inactive to source setup files.
Command - `killall gzserver`

Also, we have discuss platform to respond your queries. Kindly stick to discuss.

**Comment by JaisonJose241** [comment]
This issue is closed as of now, as there already exists a forum for the same.

**Comment by serdonke** [comment]
@architjain19, I have tried the troubleshooting steps, also `kill` takes a PID iirc, that command does not work, unless I am using zsh which autocompletes PID from name, there is no gzserver running though, I checked with htop and stuff, after I SIGINT the process. And gz server and client launches normally according to the logs.

Well, when I launch gazebo beforehand and then launch the script again, gazebo launches up with all the models and stuff, I mean gazebo launches in this case if Gazebo is already running, the command line still throws the same error.

That the `/spawn entity` service is missing or some such.
